src/utils.ts

```diff
@@ -245,13 +245,9 @@ export namespace Utils {
           "The transaction has operations that are unrecognized",
         );
       }
-      if (op.value === undefined) {
-        throw new InvalidSep10ChallengeError(
-          "The transaction's operation values should not be null",
-        );
-      }
       if (
         op.name === "web_auth_domain" &&
+        op.value &&
```

Comment:
❗ This doesn't quite do what we need it to, because if a manage data operation with key `web_auth_domain` is provided without a value, it's going to be skipped by it should error in my opinion because the SEP states that any manage data operation with that key has a value.

I think you might need something like:
```javascript
if (op.name == "web_auth_domain") {
  if (op.value === undefined) {
    throw new InvalidSep10ChallengeError("The transaction's operation values should not be null");
  }
  if (op.value.compare(Buffer.from(webAuthDomain)) {
    throw new InvalidSep10ChallengeError(`'web_auth_domain' operation value does not match ${webAuthDomain}`);
  }
}

Reply:
Very true, updating